### ISSUE TITLE
Alternatives to using a DB

### ISSUE THREAD

**Issue by sebastian-nehrdich** [ayya-vimala/pali-networks#69]
It struck me like a lightning during my meditation in the morning that we don't need to bring in a DB if we apply the following chain of tricks on text-view:
- at the moment, we have the root text and the parallels all combined into one JSON. We can split this - remove the parallels from the JSON and store them in a seperate file. In the JSON with the root text, we just give a flag has_parallel which can either be 1 or 0. If 1, we render the text blue and if 0, we don't highlight it.
- In this way, we can render the text on the left side way faster because we just need to load a small JSON 
- After the root text on the left has finished rendering (the user can now already see it and scroll it etc.), we load the JSON with the parallels into a dictionary. Now if a user clicks on a segment on the left where has_parallel is 1, we will look for the segment-nr in the dictionary with the parallels and pull the results. Only after the click has happened will we start to render the parallels in the middle. 
- additionally, but not necessary: if performance should still be problematic, we can further speed up things by bucketing the JSON with the parallels. So while generating the JSONs, I can generate them in batches of for example 1000 root segments. If the user opens a text, we load only the first 1000 segments. If the user decides to scroll down and click on a segment that lies outside of the first 1000, we apply the following logic: if has_parallel is 1, but segment-nr is not in the keys of the dictionary with the quotes, we load the JSON of the current bucket. So if we are looking at segment-nr 2300 and it is not yet in the dictionary, we have to load quotes-bucket-2000-3000.json and add the keys to the dictionary with the quotes.

This is all simple and straight forward to implement. I could sit down on the weekend and rewrite the code.

**Comment by sebastian-nehrdich** [comment]
I am already starting to rewrite the view-functions and testing out how we can do this. I will also upload some examples for split JSON files so you can have an idea on how I imagine to do this in practice.

**Comment by sebastian-nehrdich** [comment]
I have now an example online: https://github.com/ayya-vimala/pali-networks/tree/master/data-incoming/bucket_example

The file T06TD4026_root.json is just the root-text without any parallels. It is identical to the previous JSONs, except for these changes:
- I deleted 'parallels'
- I added 'has_parallel' which is either 1 or 0 
- I added 'bucket' which identifies in wich bucket we need to look in order to get the parallels. 

The folder T06TD4026_parallels contains the buckets for the parallels. 
each bucket consists of a dictionary where each segment from the root-text is a key and the value is a list of parallels. So we get all possible quotations for one segment by querying that dictionary with the segment as key. 

The way we need to change the logic in the text-view is thus: 
- after the user clicks on a segment with parallels ('has_parallel':1), we lookup if the key for the requested  segment is in the parallels-dictionary 
- if it is not yet in the dictionary, we use the value of 'bucket' in order to fetch the right bucket that contains the parallel and add the content of that bucket to the parallels-dictionary 
- if the key is already present in the dictionary, we don't need to load a bucket because it has already been loaded - fine. 

I think altogether these changes sound more dramatic than they are in practice. It adds a few lines to the Javascript but compared to the complexity of involving a database, this is a piece of cake I believe.
